uniffi_macros/src/object.rs

```diff
@@ -138,6 +143,43 @@ pub(crate) fn interface_impl(ident: &Ident, tag: Option<&Path>) -> TokenStream {
                 .concat_str(#name)
                 .concat_bool(false);
         }
+
+        unsafe #impl_spec {
+            type FfiType = <Self as ::uniffi::FfiConverterArc<#uniffi_tag>>::FfiType;
+            type ReturnType = <Self as ::uniffi::FfiConverterArc<#uniffi_tag>>::ReturnType;
+            type FutureCallback = <Self as ::uniffi::FfiConverterArc<#uniffi_tag>>::FutureCallback;
+
+            fn lower(obj: Self) -> Self::FfiType {
+                <Self as ::uniffi::FfiConverterArc<#uniffi_tag>>::lower(::std::sync::Arc::new(obj))
+            }
+
+            fn write(obj: Self, buf: &mut Vec<u8>) {
+                <Self as ::uniffi::FfiConverterArc<#uniffi_tag>>::write(::std::sync::Arc::new(obj), buf)
+            }
+
+            fn lower_return(v: Self) -> ::std::result::Result<Self::FfiType, ::uniffi::RustBuffer> {
+                <Self as ::uniffi::FfiConverterArc<#uniffi_tag>>::lower_return(::std::sync::Arc::new(v))
+            }
+
+            fn try_lift(v: Self::FfiType) -> ::uniffi::Result<Self> {
+                panic!("Can't lift")
```

Comment:
This is the main issue with this change.  If we implement `FfiConverter` on a type, then UniFFI-exported functions can input that type as an argument.  However, it's not possible to input the object type directly if it's already wrapped in an arc ([into_inner](https://doc.rust-lang.org/std/sync/struct.Arc.html#method.into_inner) doesn't seem right).

Right now this is just implemented as a panic, which is unfortunate because if a library does try to input an object type it would be better to catch the issue at compile time.  I don't think there's a static assertion to check that a function is never called though.

One solution would be to split up `FfiConverter` into 2 traits: one for lifting and one for lowering.  That seems like a big change though, is there anything easier?